jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitParseProcessor.java

```diff
@@ -89,6 +92,12 @@ private void handleJarDeployment(DeploymentPhaseContext phaseContext) throws Dep
             List<PersistenceUnitMetadataHolder> listPUHolders = new ArrayList<PersistenceUnitMetadataHolder>(1);
             // handle META-INF/persistence.xml
             final ResourceRoot deploymentRoot = deploymentUnit.getAttachment(Attachments.DEPLOYMENT_ROOT);
+            VirtualFile appclientClientXml = deploymentRoot.getRoot().getChild(META_INF_APPLICATION_CLIENT_XML);
+            if (!appclient && appclientClientXml.exists()) {
```

Comment:
I think this should check the DeploymentTypeMarker from Attachments.DEPLOYMENT_TYPE, as this is an incomplete check vs what ApplicationClientDeploymentProcessor does. And making it complete would be duplicative

If Attachments.DEPLOYMENT_TYPE isn't giving you what's needed we should investigate why not. AFAICT it should.

Reply:
In zulip chat you mentioned it it wasn't set the way you expected, so I thought there might be an issue. Maybe I misunderstood. We'll see. :)

Reply:
Thanks Brian!  Change is added for ^

> In zulip chat you mentioned it it wasn't set the way you expected, so I thought there might be an issue. Maybe I misunderstood. We'll see. :)

I was read the words you said but I tried using a marker that the appclient container subsystem sets but that I likely used the wrong marker (probably one that isn't set in the server).

Reply:
Thanks Brian! Change is added for ^